### PR TITLE
Add has_ctrl_influence primitive to Context

### PR DESCRIPTION
**Pull request by livia01px2019**
## What Changed?

Add a new primitive that checks whether a src has data then ctrl flow influence on a sink.

## Why Does It Need To?

Common pattern when we talk about control flow influence.

## Checklist

- [x] Above description has been filled out so that upon quash merge we have a
  good record of what changed.
- [x] New functions, methods, types are documented. Old documentation is updated
  if necessary
- [ ] Documentation in Notion has been updated
- [x] Tests for new behaviors are provided
  - [ ] New test suites (if any) ave been added to the CI tests (in
    `.github/workflows/rust.yml`) either as compiler test or integration test.
    *Or* justification for their omission from CI has been provided in this PR
    description.

### CODE REVIEW

**Review by carolynzech — APPROVED** [review]

File: crates/paralegal-policy/src/context.rs [hunk 1]
```diff
@@ -299,6 +299,8 @@ impl Context {
     /// Returns whether a node flows to a node through the configured edge type.
     ///
     /// Nodes do not flow to themselves. CallArgument nodes do flow to their respective CallSites.
+    ///
+    /// If you use flows_to with [`EdgeType::Control`], you might want to consider using has_ctrl_influence.
```
Comment (JustusAdam) [inline crates/paralegal-policy/src/context.rs:303]: Please explain briefly why someone would want to use `has_ctrl_influence` instead (what is the difference between that and using `EdgeType::Control` on this function).

Also please use a reference to `has_ctrl_influence` the same way we reference `EdgeType::Control`.

File: crates/paralegal-policy/src/context.rs [hunk 2]
```diff
@@ -345,6 +347,18 @@ impl Context {
         }
     }
 
+    /// Returns whether there is direct control flow influence from influencer to sink, or is there is some data flow from influencer to something that has ctrl flow to sink.
```
Comment (JustusAdam) [inline crates/paralegal-policy/src/context.rs:350]: I would say "or there is some node `n`, which is data-flow influenced by `influencer` and has direct control flow influence on `target`. Or as expressed in code: some `n` such that `self.flows_to(influencer, n, EdgeType::Data) && self.flows_to(n, target, Edgetype::Control)`."

**Review by JustusAdam — APPROVED** [review]

File: crates/paralegal-policy/src/context.rs (hunk 1 above)
Comment (livia01px2019) [reply crates/paralegal-policy/src/context.rs:303]: done

File: crates/paralegal-policy/src/context.rs (hunk 2 above)
Comment (livia01px2019) [reply crates/paralegal-policy/src/context.rs]: done

**Pull request merged**